dockerfiles/depwatcher/src/depwatcher/app_dynamics_agent.cr

```diff
@@ -66,8 +67,14 @@ module Depwatcher
     getter patch : Int32
     getter metadata : Int32
 
+    # AppDynamics uses a different versioning scheme than SemVer. They use calendar versioning (ref https://community.appdynamics.com/t5/Knowledge-Base/New-in-March-2020-AppDynamics-is-switching-to-calendar/ta-p/38364)
+    # So every new release will follow the same pattern: YY.M.P-X  (YY = year, M = month, P = patch, X = metadata). Example: 22.1.0-14
     def initialize(@original : String)
-      @major, @minor, @patch, @metadata = @original.split(".").map(&.to_i)
+      splitVersion = @original.split(".")
+      @major = splitVersion[0].to_i
+      @minor = splitVersion[1].to_i
+      @patch = splitVersion[2].split("-")[0].to_i
+      @metadata = splitVersion[2].split("-")[1].to_i
```

Comment:
Consider storing `splitVersion[2].split("-")` rather than retyping, as it seems like something that could get out of sync easily